packages/upload/theme/material/vaadin-upload-styles.js

```diff
@@ -179,6 +180,10 @@ registerStyles(
       color: inherit;
     }
 
+    [part$='button'][disabled] {
```

Comment:
The Lumo theme includes `field-button.js` which automatically adds disabled styles for all buttons. For Material custom styles are needed.